File: app/widgets/molecular/dna_samples/taxon_combo_box.rb
class Molecular::DnaSamples::TaxonComboBox < Templates::ComboBox
  def initialize options
    @dna_sample ||= options[:dna_sample] || fail('dna sample not given')
    @model_object = @dna_sample
    @attribute_path = 'taxon'
    @value_method = 'taxon_id'
    @text_method = 'label'
    @search_method = 'name'
    @width = 325
    super
  end

  def value_searchtext
    #(value && ((value.respond_to?(value_method) && value.send(value_method)) || value[search_method])) || ''
    
    @dna_sample.try('taxon').try('name') || ''
  end

  def value_fulltext
    
    #(value && ((value.respond_to?(value_method) && value.send(value_method)) || value[search_method])) || ''
    @dna_sample.try('taxon').try('name') || ''
  end
end

File: app/widgets/molecular/dna_samples/taxon_combo_box.js
//= require <templates/combo_box>
//= require <sync_record>

JooseModule('Molecular.DnaSamples', function () {
  JooseClass('TaxonComboBox', {
    isa: Templates.ComboBox,
    has: {
      object:  { is: 'ro', required: true, nullable: false },
      method:  { is: 'ro', init: 'taxon' },
      collectionURI: { is: 'ro', init: function () { return Route.forPathname('project_taxa_path') } },
      valueMethod:   { is: 'ro', init: 'taxon_id' },
      textMethod:    { is: 'ro', init: 'label' },
      searchMethod:  { is: 'ro', init: 'name' },
      width:         { is: 'ro', init: 325 },
      options: { is: 'ro', init: function () { return {
        parameters: {
          joins:      'namestatus',
          include:    'namestatus',
          order:      'name',
          select:     'taxon_id,name,label,namestatus_id',
//          select:     'taxon_id,name,label',  //previously some taxa records had null namestatus id but fixed with migration 20120910161627
          limit:      '100'
      }}}},
      tagName: { init: 'dna_sample' }
    }
  })
});
